### ISSUE TITLE
Fix request: AvoidInfiniteRecursion false positive other types in method call

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#507]
```java
public class Foo {

    private final Map<ComponentType, List<Descriptor>> coreMap = new HashMap<>();
    private final Map<ComponentType, List<Descriptor>> edgeMap = new HashMap<>();
    
    public List<Descriptor> getComponents(Set<ComponentType> types, RuleType ruleType) {
        if (ruleType.CORE.equals(ruleType)) {
            return getComponents(types, coreMap); // <- false positive
        } else if (ruleType.EDGE.equals(ruleType)) {
            return getComponents(types, edgeMap); // <- false positive
        } else {
            throw new RuntimeException("Unsupported rule type " + ruleType);
        }
    }

    private List<Descriptor> getComponents(Set<ComponentType> types, Map<ComponentType, List<Descriptor>> componentsMap) {
        List<Descriptor> result = new ArrayList<>();
        // some logic here
        return Collections.unmodifiableList(result);
    }
}
```

**Comment by stokpop** [comment]
There are still many false positives with methods with same number of arguments but other types (e.g. more than half of the 23 violations in a test project).

Is it an idea to try and assume it is _no_ recursion when another method is found in same class with same number of params?
